### PR TITLE
Fix for preferred address list

### PR DESCRIPTION
**Pull request by RostislavReznichenko**
InetUtils doesn't work correctly with list of preferred addresses because it continue iterate throughs list of preferred addresses even when predicate for preferred address gives positive result and stop iterate on negative result only.  

So we have 2 cases:

- Preferred addresses are [192, 10], address for test is 10.x.x.x
`!address.getHostAddress().matches("10") && !address.getHostAddress().startsWith("10")` will pass and iteration will be stoped.
- Preferred addresses are [192, 10], address for test is 192.x.x.x
In first iteration 
`!address.getHostAddress().matches("192") && !address.getHostAddress().startsWith("192)` will not pass and iteration will continue.
In second iteration 
`!address.getHostAddress().matches("10") && !address.getHostAddress().startsWith("10)` will pass and iteration will be stoped with invalid result.

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@RostislavReznichenko Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-cloud/spring-cloud-commons&pullRequestId=332)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-cloud/spring-cloud-commons&pullRequestId=332) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@RostislavReznichenko Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-cloud/spring-cloud-commons&pullRequestId=332)!

File: spring-cloud-commons/src/main/java/org/springframework/cloud/commons/util/InetUtils.java [hunk 1]
```diff
@@ -24,15 +27,8 @@
...
-import java.util.concurrent.ExecutorService;
-import java.util.concurrent.Executors;
-import java.util.concurrent.Future;
-import java.util.concurrent.ThreadFactory;
-import java.util.concurrent.TimeUnit;
-
-import org.apache.commons.logging.Log;
-import org.apache.commons.logging.LogFactory;
+import java.util.List;
+import java.util.concurrent.*;
```
Comment (ryanjbaxter) [inline spring-cloud-commons/src/main/java/org/springframework/cloud/commons/util/InetUtils.java]: No `*` imports

**Comment by RostislavReznichenko** [comment]
@ryanjbaxter done

File: spring-cloud-commons/src/test/java/org/springframework/cloud/commons/util/InetUtilsTests.java [hunk 2]
```diff
@@ -16,15 +16,14 @@
...
 import org.springframework.cloud.commons.util.InetUtils.HostInfo;
 
-import static org.junit.Assert.assertFalse;
-import static org.junit.Assert.assertNotNull;
-import static org.junit.Assert.assertTrue;
+import java.net.InetAddress;
+import java.util.Arrays;
+import java.util.Collections;
+
+import static org.junit.Assert.*;
```
Comment (ryanjbaxter) [inline spring-cloud-commons/src/test/java/org/springframework/cloud/commons/util/InetUtilsTests.java]: Another `*` import

**Comment by RostislavReznichenko** [comment]
@ryanjbaxter done

**Comment by spencergibb** [comment]
Squashed and merged with some polish via f94c75f0df841fb8f765f94d2d1698f7fcc0e8f9

**Pull request closed without merging**